src/lib/mongoBackend/dbFieldEncoding.cpp

```diff
@@ -41,8 +41,9 @@
 * Replace:
 *   . => =
 */
-std::string dbDotEncode(std::string s)
+std::string dbDotEncode(const std::string& _s)
 {
+  std::string s = _s; // replace cannot be used in const std::string&
   std::replace(s.begin(), s.end(), ESCAPE_1_DECODED, ESCAPE_1_ENCODED);
   return s;
 }
```

Comment:
Copy + modify + return.

(Note that previous versions of this function also return the modified string... probably previous design wasn't correct, given it was somewhere "in the middle" of both possiblities; what I wonder is how it could work before :).